### PR TITLE
IDEX-3984: Rework client side PT framework

### PR DESCRIPTION
**Pull request by gazarenkov**

### CODE REVIEW

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java [hunk 1]
```diff
@@ -44,17 +42,24 @@ public GitAction(String text,
...
+//
+//            List<String> listVcsProvider = rootProjectDescriptor.getAttributes().get("vcs.provider.name");
+//
+//            if (listVcsProvider != null && (!listVcsProvider.isEmpty()) && listVcsProvider.contains("git")) {
+//                isGitRepository = true;
+//            }
+//
+//        }
+//        return isGitRepository;
     }
```
Comment (dimasnurenko) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java]: Can you delete commented code?

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java [hunk 2]
```diff
@@ -44,17 +42,24 @@ public GitAction(String text,
...
-        if (getActiveProject() != null) {
-            ProjectConfigDto rootProjectDescriptor = getActiveProject().getRootProject();
-            List<String> listVcsProvider = rootProjectDescriptor.getAttributes().get("vcs.provider.name");
 
-            if (listVcsProvider != null && (!listVcsProvider.isEmpty()) && listVcsProvider.contains("git")) {
-                isGitRepository = true;
-            }
-        }
-        return isGitRepository;
+        return appContext.getActiveProject().isTypeOf("git");
```
Comment (vparfonov) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java]: need NPE checking, appContext.getActiveProject() can be null

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java (hunk 1 above)
Comment (gazarenkov) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java]: Yes, sure, but I'd like to make sure all works as expected first.

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java (hunk 2 above)
Comment (gazarenkov) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java]: done

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java [hunk 3]
```diff
@@ -44,17 +42,26 @@ public GitAction(String text,
     }
 
     protected boolean isGitRepository() {
-        boolean isGitRepository = false;
 
-        if (getActiveProject() != null) {
-            ProjectConfigDto rootProjectDescriptor = getActiveProject().getRootProject();
-            List<String> listVcsProvider = rootProjectDescriptor.getAttributes().get("vcs.provider.name");
+        if(appContext.getActiveProject() == null)
```
Comment (azatsarynnyy) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/GitAction.java]: *if*-block should be formatted in accordance with codestyle: https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#non-empty-blocks

**Pull request closed without merging**